src/garage/np/policies/policy.py

```diff
@@ -20,77 +12,84 @@ def get_action(self, observation):
         Args:
             observation (np.ndarray): Observation from the environment.
 
-        Returns:
-            (np.ndarray): Action sampled from the policy.
+        """
+
+    @abc.abstractmethod
+    def get_actions(self, observations):
+        """Get actions given observations.
+
+        Args:
+            observations (torch.Tensor): Observations from the environment.
 
         """
 
-    def reset(self, dones=None):
+    def reset(self, do_resets=None):
         """Reset the policy.
 
-        If dones is None, it will be by default np.array([True]) which implies
-        the policy will not be "vectorized", i.e. number of parallel
-        environments for training data sampling = 1.
+        This is effective only to recurrent policies.
+
+        do_resets is an array of boolean indicating
+        which internal states to be reset. The length of do_resets should be
+        equal to the length of inputs, i.e. batch size.
 
         Args:
-            dones (numpy.ndarray): Bool that indicates terminal state(s).
+            do_resets (numpy.ndarray): Bool array indicating which states
+                to be reset.
 
         """
 
     @property
     def observation_space(self):
         """akro.Space: The observation space of the environment."""
-        return self._env_spec.observation_space
 
     @property
     def action_space(self):
         """akro.Space: The action space for the environment."""
-        return self._env_spec.action_space
 
-    def log_diagnostics(self, paths):
-        """Log extra information per iteration based on the collected paths.
+    @property
+    def env_spec(self):
+        """Policy environment specification.
 
-        Args:
-            paths (list[dict]): A list of collected paths
+        Returns:
+            garage.EnvSpec: Environment specification.
 
         """
 
     @property
-    def state_info_keys(self):
-        """Get keys describing policy's state.
+    def name(self):
+        """Name of policy.
 
         Returns:
-            List[str]: keys for the information related to the policy's state
-            when taking an action.
+            str: Name of policy
 
         """
-        return list()
 
-    def terminate(self):
-        """Clean up operation."""
+    def log_diagnostics(self, paths):
+        """Log extra information per iteration based on the collected paths.
 
+        Args:
+            paths (list[dict]): A list of collected paths
 
-class StochasticPolicy(Policy):
-    """Base class for stochastic policies implemented in numpy."""
+        """
 
     @property
-    @abc.abstractmethod
-    def distribution(self):
-        """Get the distribution of the policy.
+    def state_info_specs(self):
```

Comment:
This is really only ever been useful in the TensorFlow branch. Is it really necessary?